spec.html

```diff
@@ -122,18 +122,33 @@ <h1>[[Call]] ( _thisArgument_, _argumentsList_ )</h1>
 	</emu-clause>
 
 	<emu-clause id="sec-wrappedfunctioncreate" aoid="WrappedFunctionCreate">
-		<h1>WrappedFunctionCreate ( _callerRealm_, _targetFunction_ )</h1>
-		<p>The abstract operation WrappedFunctionCreate takes arguments _callerRealm_ and _targetFunction_. It is used to specify the creation of new wrapped function exotic objects. It performs the following steps when called:</p>
+		<h1>WrappedFunctionCreate ( _callerRealm_, _Target_ )</h1>
+		<p>The abstract operation WrappedFunctionCreate takes arguments _callerRealm_ and _Target_. It is used to specify the creation of new wrapped function exotic objects. It performs the following steps when called:</p>
 		<emu-alg>
 			1. Assert: _callerRealm_ is a Realm Record.
-			1. Assert: IsCallable(_targetFunction_) is *true*.
+			1. Assert: IsCallable(_Target_) is *true*.
 			1. Let _internalSlotsList_ be the internal slots listed in <emu-xref href="#table-internal-slots-of-wrapped-function-exotic-objects"></emu-xref>, plus [[Prototype]] and [[Extensible]].
-			1. Let _obj_ be ! MakeBasicObject(_internalSlotsList_).
-			1. Set _obj_.[[Prototype]] to _callerRealm_.[[Intrinsics]].[[%Function.prototype%]].
-			1. Set _obj_.[[Call]] as described in <emu-xref href="#sec-wrapped-function-exotic-objects-call-thisargument-argumentslist"></emu-xref>.
-			1. Set _obj_.[[WrappedTargetFunction]] to _targetFunction_.
-			1. Set _obj_.[[Realm]] to _callerRealm_.
-			1. Return _obj_.
+			1. Let _wrapped_ be ! MakeBasicObject(_internalSlotsList_).
+			1. Set _wrapped_.[[Prototype]] to _callerRealm_.[[Intrinsics]].[[%Function.prototype%]].
+			1. Set _wrapped_.[[Call]] as described in <emu-xref href="#sec-wrapped-function-exotic-objects-call-thisargument-argumentslist"></emu-xref>.
+			1. Set _wrapped_.[[WrappedTargetFunction]] to _Target_.
+			1. Set _wrapped_.[[Realm]] to _callerRealm_.
+			1. Let _L_ be 0.
+			1. Let targetHasLength be ? HasOwnProperty(_Target_, *"length"*).
+			1. If _targetHasLength_ is *true*, then
+				1. Let _targetLen_ be ? Get(_Target_, *"length"*).
+				1. If Type(_targetLen_) is Number, then
+					1. If _targetLen_ is *+&infin;*<sub>𝔽</sub>, set _L_ to +&infin;.
+					1. Else if _targetLen_ is *-&infin;*<sub>𝔽</sub>, set _L_ to 0.
+					1. Else,
+					1. Let _targetLenAsInt_ be ! ToIntegerOrInfinity(_targetLen_).
+						1. Assert: _targetLenAsInt_ is finite.
+						1. Set _L_ to max(_targetLenAsInt_, 0).
+			1. Perform ! SetFunctionLength(_wrapped_, _L_).
+			1. Let _targetName_ be ? Get(_Target_, *"name"*).
+			1. If Type(_targetName_) is not String, set _targetName_ to the empty String.
```

Comment:
@ljharb this is interesting! `Function#bind` normalizes to only copy string names, so I also applied this change here as you mentioned in the comments above.

There is a _catch_ where we need to avoid bringing error objects from the target function and that can happen on the HasOwnProperty and Get calls.

I believe that doing the common abstraction - as you suggested - would make it easier to handle this. I'll follow up with a new commit soon.